testsuite/features/step_definitions/command_steps.rb

```diff
@@ -9,12 +9,10 @@
 
 Then(/^"([^"]*)" should have a FQDN$/) do |host|
   node = get_target(host)
-  result_1, return_code_1 = node.run('hostname -f')
-  result_1.delete!("\n")
-  result_2, return_code_2 = node.run('python -c "import socket; print socket.getfqdn()"')
```

Comment:
Not sure about it. This way of finding the fqdn is used for traditional clients and i remember a case where hostname -f was able to return a fqdn and socket.getfqdn not.

Maybe checking if python or python3 is available and execute the test in this case only? Otherwise the hostname -f is enough.

What do you think?

Reply:
I am not 100 % sure, but I seem to remember `socket.getfqdn()` not giving same result as `hostname -f` was a python problem, not a network (infra) issue. And it happened only one time.

I'd like these sanity checks to remain sanity checks, infra oriented and not trying to reproduce all past software bugs.

From a networking POV the valid test is hostname -f, because it tests /etc/hostname, direct, and reverse resolution, all at once.